src/test/java/com/commercetools/project/sync/CliRunnerTest.java

```diff
@@ -429,6 +456,8 @@ void run_WithSyncAsArgumentWithAllArg_ShouldExecuteAllSyncers() {
         .thenReturn(CompletableFuture.completedFuture(PagedQueryResult.empty()));
     when(sourceClient.execute(any(CartDiscountQuery.class)))
         .thenReturn(CompletableFuture.completedFuture(PagedQueryResult.empty()));
+    when(sourceClient.execute(any(StateQuery.class)))
+        .thenReturn(CompletableFuture.completedFuture(PagedQueryResult.empty()));
```

Comment:
shall move this section few lines above where it mock as 3rd query? :) Just to be consistent.

Reply:
I don't understand what you mean?

Reply:
Doesn't make sense :) The order doesn't matter here :) I think no one really cared about it here :)

Reply:
But when I was reading this method only this part was looking odd. It's just matter of moving 2 lines of code above. Not much work :D

Reply:
Why does it look odd? :) It's not much work, but I want to understand why do you think it makes more sense when here it actually doesn't matter at all? :)

Reply:
I changed it a bit to make it even more obvious that order doesn't matter. https://github.com/commercetools/commercetools-project-sync/pull/133/commits/77bc069cfa9986ea6fd88df77dfb3dbf0398fd72

Reply:
I see it as cognitive complexity on the test cases. `ProductTypeSyncer`, `TypeSyncer`, `StateSyncer` have the first priority over the other Syncer classes in project sync execution. This intuition has been preserved in all integration test mockings. A developer who goes through the test cases without even looking at the source code will get some sense on the priorities of Sync classes, but with this change that intuition will be broken. 

>I changed it a bit to make it even more obvious that order doesn't matter

Yes, order doesn't matter for mocking in this case, but have you noticed that `ProductTypeQuery` or `TypeQuery` was always created at the beginning of the mocking statements? I would still vote to move `StateQuery` mocking next to  `TypeQuery` whenever it is being mocked inside a test case.

Reply:
You might be right if it was like this before :) But it was not.

This is the order that we do the sync:
```
final List<CompletableFuture<Void>> typeAndProductTypeAndStateSync =
        asList(
            ProductTypeSyncer.of(sourceClient, targetClient, clock)
                .sync(runnerNameOptionValue, isFullSync)
                .toCompletableFuture(),
            TypeSyncer.of(sourceClient, targetClient, clock)
                .sync(runnerNameOptionValue, isFullSync)
                .toCompletableFuture(),
            StateSyncer.of(sourceClient, targetClient, clock)
                .sync(runnerNameOptionValue, isFullSync)
                .toCompletableFuture());

    return CompletableFuture.allOf(typeAndProductTypeAndStateSync.toArray(new CompletableFuture[0]))
        .thenCompose(
            ignored ->
                CategorySyncer.of(sourceClient, targetClient, clock)
                    .sync(runnerNameOptionValue, isFullSync))
        .thenCompose(
            ignored ->
                ProductSyncer.of(sourceClient, targetClient, clock)
                    .sync(runnerNameOptionValue, isFullSync))
        .thenCompose(
            ignored ->
                CartDiscountSyncer.of(sourceClient, targetClient, clock)
                    .sync(runnerNameOptionValue, isFullSync))
        .thenCompose(
            ignored ->
                InventoryEntrySyncer.of(sourceClient, targetClient, clock)
                    .sync(runnerNameOptionValue, isFullSync))
```

It goes from Category -> Product -> CardDiscount -> InventoryEntry. The test was not like that even before my change :) So that's why I don't see your point. Additionally your point might add more confusion for people to think that there must be an order like in the implementation, which is not true.

Reply:
>It goes from Category -> Product -> CardDiscount -> InventoryEntry. The test was not like that even before my change :)

Yes, these are the 2 phase syncers. Order doesn't matter here. But when mocking, `ProductTypeSyncer` & `TypeSyncer` is always mocked at the first. Because it convey the message about the priority of syncers. And it is even specifically mentioned in the JIRA ticket about the execution order for StateSync. 

> Additionally your point might add more confusion for people to think that there must be an order like in the implementation, which is not true.

As per my understanding, `ProductTypeSyncer `, `TypeSyncer` & `StateSyncer` will be executed. Once all of those 3 types are finished syncing, we move to rest of the syncers.

You think this message should not be reflected in test mocking? 

(I'll approve the PR as we are time restricted on our work on this ticket & it doesn't affect the real execution of the project-sync)

Reply:
No, I think there's no such rule that mocking should reflect the current implementation. I think in ideal case this test should not care at all how it is implemented underneath as long as the expected results correctly returned. If we apply the rule you wrote, then when I want to change the order of execution, I also have to change this mocking ting for no reason, even if the tests are passing. That's why I don't think tests should care about the implementation underneath.